src/main/java/dev/openfeature/javasdk/ProviderEvaluation.java

```diff
@@ -9,6 +9,7 @@
 public class ProviderEvaluation<T> implements BaseEvaluation<T> {
```

Comment:
Would it be useful to have an overload for setting the reason based on a Reason? I am not really sure how these lombok classes work, aside from magic.

Reply:
I'm hesitant to add it b/c it's not something lombok can do for us. I'm lazy.